### PR TITLE
Add more support for tags

### PR DESCRIPTION
**Pull request by clns**
Add `TagsCollection`.

Add support for `git_tag_list`, `git_tag_list_match`, `git_tag_foreach`, `git_tag_create_lightweight`.

### CODE REVIEW

**Comment by carlosmn** [comment]
This moves the code back to having a lot of methods on `Repository`, which is a style which current git2go has abandoned. These should be methods on a `TagsCollection` object.

File: tag.go [hunk 1]
```diff
@@ -42,3 +48,129 @@ func (t Tag) TargetId() *Oid {
...
+// The name of the tag is validated for consistency (see git_tag_create() for the rules
+// https://libgit2.github.com/libgit2/#HEAD/group/tag/git_tag_create) and should
+// not conflict with an already existing tag name.
+//
+// If force is true and a reference already exists with the given name, it'll be replaced.
+//
+// The created tag is a simple reference and can be queried using
+// repo.References.Lookup("refs/tags/<name>"). The name of the tag (eg "v1.0.0")
+// is queried with ref.Shorthand().
+func (repo *Repository) CreateTagLightweight(name string, commit *Commit, force bool) (*Oid, error) {
```
Comment (carlosmn) [inline tag.go]: This needs to explain what the return Oid is, there is no way to know what it is from the signature.

Comment (clns) [inline tag.go]: I added more explanation in the next commit.

**Comment by clns** [comment]
I noticed the other collections but since there wasn't any for tags I didn't know if I should add one. I can go ahead and add a `TagsCollection`, let me know.

**Comment by carlosmn** [comment]
Yes, do add `TagsCollection`.

**Comment by clns** [comment]
Added `TagsCollection` and moved the new methods on it.

This PR doesn't break any existing apis, so moving the existing tags-related methods from `Repository` into `TagsCollection` is outside of the scope of this PR and is a todo.

**Pull request merged**